pkg/chains/storage/oci/oci.go

```diff
@@ -194,9 +196,60 @@ func (b *Backend) Type() string {
 }
 
 func (b *Backend) RetrieveSignature(opts config.StorageOpts) (string, error) {
-	return "", fmt.Errorf("not implemented")
+	sigs, err := b.RetrieveArtifact(opts)
+	if err != nil {
+		return "", err
+	}
+
+	for _, s := range sigs {
+		// also update this once the function def is updated
+		if sig, err := s.Base64Signature(); err == nil {
+			return sig, nil
+		}
+	}
+	return "", fmt.Errorf("failed to find a signature")
 }
 
 func (b *Backend) RetrievePayload(opts config.StorageOpts) (string, error) {
-	return "", fmt.Errorf("not implemented")
+	sigs, err := b.RetrieveArtifact(opts)
+	if err != nil {
+		return "", err
+	}
+
+	for _, s := range sigs {
+		if payload, err := s.Payload(); err == nil {
+			return string(payload), nil
+		}
+	}
+	return "", fmt.Errorf("failed to find a payload")
+
+}
+
+func (b *Backend) RetrieveArtifact(opts config.StorageOpts) ([]oci.Signature, error) {
+	// given the TaskRun, retrieve the signature
+	images := artifacts.ExtractOCIImagesFromResults(b.tr, b.logger)
+
+	// we can remove this check once the function def is changed
+	if len(images) != 1 {
+		return nil, errors.New("multiple images found, unsure which to provide signature for")
+	}
+	// get signature for a specific image
+	ref, ok := images[0].(name.Digest)
+	if !ok {
+		return nil, errors.New("error parsing image")
+	}
+	img, err := ociremote.SignedImage(ref)
+	if err != nil {
+		return nil, err
+	}
+	sigImage, err := img.Signatures()
```

Comment:
Hey @rgreinho, I believe if you want to get the payload for the attestation, you'll have to call `img.Attestations()` instead of `img.Signatures()` here, and then call `Payload()` on the output of that!

Reply:
It worked 🤗